bson/src/main/org/bson/ByteBufNIO.java

```diff
@@ -35,7 +35,7 @@
      * @param buf the {@code ByteBuffer} to wrap.
      */
     public ByteBufNIO(final ByteBuffer buf) {
-        this.buf = buf;
+        this.buf = buf.order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
I did something similar in the OP_COMPRESSED PR.  Doing it here as well allows header parsing to simply use ByteBuf instead of ByteBufferBsonInput.